### DISCUSSION TITLE
Negative values in both predicted and observed plot of GLM model fit.

### DISCUSSION THREAD

**Discussion by DaniyalBME** [bachlab/PsPM#701] [Q&A] [answered]
Hi everyone,

I hope this message finds you all well. I'm new to using PSPM and recently applied a GLM model to a skin conductance data. I organized the data according to the onset of markers, but I'm encountering an issue with negative values in both predicted and observed plots of model fit (see attached figure).

![GLM_Model](https://github.com/bachlab/PsPM/assets/168282904/3cea348a-7012-4dc6-8925-a458db446278)

Could anyone please advise on how to address this problem? I have attached my data files for reference.

[SCR Files.zip](https://github.com/bachlab/PsPM/files/15139223/SCR.Files.zip)

Thank you for your help!

Best regards,
Daniyal

**Comment by dominikbach** [comment, marked as answer]
Hi Danyal
apologies for the late reply, and thanks for providing the data.
I wrote a rough script to extract the actually elicited responses per condition (see below) , and this is what I get:
![image](https://github.com/bachlab/PsPM/assets/29309412/70ae1ae5-1b51-41cb-b2cb-20b939da94a2)
Only one of your conditions (black) appears to elicit an average SCR, and even this one does not look like a canonical SCR (overlaid in red). So, the negative parameter estimates reflect your data in at least two conditions - in the others they might be caused by collinearity between the regressors (see the orthogonality plot in the review model dialogue).
It might be good to check whether this is also the case in the original data, and the timing definition.
Hope this helps.
Dominik
```
[sts, segments] = pspm_extract_segments('model', 'GLM_PD2.mat', struct());
close all
for k = 1:4
    plotdata = segments.segments{k}.mean;
    plot(segments.segments{k}.t, plotdata - plotdata(1), 'Color', 2 * k * [.1, .1, .1])
    hold on
end
plotdata = pspm_bf_scrf_f(1/1000);
plot(segments.segments{k}.t, plotdata(1:numel(segments.segments{k}.t))*range(segments.segments{1}.mean), 'r');
```
* (the first line will work in the develop branch of PsPM, not in 6.1.2)
